### PR TITLE
feat: add dotnet aot build

### LINKED ISSUE #2: .NET 7 Native AOT
Would be interesting to see results with .NET 7 Lambda functions with Native AOT compilation (also with `provided.al2` runtime)
https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/build-dotnet7.html

to demystify the believe in `x10` performance boost - https://twitter.com/maloo321/status/1619788488316170240

### PR DESCRIPTION
**Pull request by RichiCoder1**
Adds a dotnet aot runtime.

This is an adaption and simplification of the AWS AOT template.

My docker is being quirky so I haven't been able to validate it.

Closes #2

### CODE REVIEW

**Comment by RichiCoder1** [comment]
Apologies if this is partially incomplete, wasn't sure what all I needed to touch!

File: function-deployer/app.js [hunk 1]
```diff
@@ -44,6 +44,11 @@ const functionsToDeploy = [
         runtime:    'dotnet6',
         handler:    'LambdaPerf::LambdaPerf.Function::Handler',
     },
+    {
+        runtime:    'provided',
```
Comment (maxday) [inline function-deployer/app.js:48]: could we use `provided.al2` instead?

File: s3-uploader/runtimes/dotnet7_aot/build.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+SOURCE=${BASH_SOURCE[0]}
+DIR_NAME=$(dirname ${SOURCE})
+rm ${DIR_NAME}/code.zip 2> /dev/null
+
+docker build ${DIR_NAME} -t maxday/dotnet6
```
Comment (maxday) [inline s3-uploader/runtimes/dotnet7_aot/build.sh:5]: I think you meant `dotnet7` ?

File: s3-uploader/runtimes/dotnet7_aot/build.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+dockerId=$(docker create maxday/dotnet6)
```
Comment (maxday) [inline s3-uploader/runtimes/dotnet7_aot/build.sh:6]: same here?

File: s3-uploader/runtimes/dotnet7_aot/index.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+exports.handler = () => {
```
Comment (maxday) [inline s3-uploader/runtimes/dotnet7_aot/index.js:1]: I think you can remove this file

**Review by maxday — CHANGES_REQUESTED** [review]
Huge thanks for your contribution 💯! That's super helpful! I left some minor comments.
Let me know:
- if you want/have time to make some adjustments
or
- if it's ok for you if I push changes to your branch so we can merge!

File: s3-uploader/runtimes/dotnet7_aot/build.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
+SOURCE=${BASH_SOURCE[0]}
```
Comment (maxday) [inline s3-uploader/runtimes/dotnet7_aot/build.sh:1]: could you `chmod u+x` this `build.sh` file please?

File: function-deployer/app.js (hunk 1 above)
Comment (RichiCoder1) [reply function-deployer/app.js:48]: Ah yup! I think that was the recommend one, my bad.

File: s3-uploader/runtimes/dotnet7_aot/build.sh (hunk 2 above)
Comment (RichiCoder1) [reply s3-uploader/runtimes/dotnet7_aot/build.sh:5]: I did, and changed this too 🤔. Must have been an issue saving, I can resolve this.

**Comment by RichiCoder1** [comment]
Either or! I'll try to get these fixed tomorrow but also if there's anything I miss def welcome any changes.

**Review by maxday — APPROVED** [review]
Thanks a lot! 💯

**Pull request merged**